0.18.0: build fails because missing `-lz` on linking

```console
[100%] Linking CXX executable bpftrace_test
cd /home/tkloczko/rpmbuild/BUILD/bpftrace-0.18.0/x86_64-redhat-linux-gnu/tests && /usr/bin/cmake -E cmake_link_script CMakeFiles/bpftrace_test.dir/link.txt --verbose=1
/usr/bin/g++ -O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-
cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clas
h-protection -fcf-protection -fdata-sections -ffunction-sections -flto=auto -flto-partition=none -Os -DNDEBUG -Wl,--gc-sections -Wl,--as-needed -flto=auto -flto-partition=none -fuse-linker-
plugin -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -Wl,--build-id=sha1 CMakeFiles/bpftrace_test.dir/ast.cpp.o CMakeFiles/bpftrace_test.dir/bpftrace.cpp.o CMakeFiles/bpftrace_te
st.dir/child.cpp.o CMakeFiles/bpftrace_test.dir/clang_parser.cpp.o CMakeFiles/bpftrace_test.dir/field_analyser.cpp.o CMakeFiles/bpftrace_test.dir/log.cpp.o CMakeFiles/bpftrace_test.dir/main
.cpp.o CMakeFiles/bpftrace_test.dir/mocks.cpp.o CMakeFiles/bpftrace_test.dir/parser.cpp.o CMakeFiles/bpftrace_test.dir/portability_analyser.cpp.o CMakeFiles/bpftrace_test.dir/procmon.cpp.o
CMakeFiles/bpftrace_test.dir/probe.cpp.o CMakeFiles/bpftrace_test.dir/resource_analyser.cpp.o CMakeFiles/bpftrace_test.dir/required_resources.cpp.o CMakeFiles/bpftrace_test.dir/semantic_ana
lyser.cpp.o CMakeFiles/bpftrace_test.dir/tracepoint_format_parser.cpp.o CMakeFiles/bpftrace_test.dir/utils.cpp.o -o bpftrace_test  ../src/libbpftrace.a /usr/lib64/libgtest_main.so.1.13.0 /u
sr/lib64/libgmock.so ../resources/libresources.a ../src/libruntime.a /usr/lib64/libbpf.so /usr/lib64/libbcc_bpf.so /usr/lib64/libelf.so ../src/aot/libaot.a ../src/ast/libast.a ../libparser.
a ../src/ast/libast_defs.a /usr/lib64/libclang.so.16.0.3 /usr/lib64/libLLVM-16.so.0 ../src/arch/libarch.a ../src/cxxdemangler/libcxxdemangler_llvm.a /usr/lib64/libgtest.so.1.13.0
/home/tkloczko/rpmbuild/BUILD/bpftrace-0.18.0/src/driver.cpp:8:14: warning: 'yy_scan_string' violates the C++ One Definition Rule [-Wodr]
    8 | extern void *yy_scan_string(const char *yy_str, yyscan_t yyscanner);
      |              ^
/home/tkloczko/rpmbuild/BUILD/bpftrace-0.18.0/x86_64-redhat-linux-gnu/lex.yy.cc:2560:17: note: 'yy_scan_string' was previously declared here
 2560 |  */
      |                 ^
/usr/bin/ld: /usr/lib64/libbcc_bpf.so: undefined reference to symbol 'gzgets'
/usr/bin/ld: /usr/lib64/libz.so.1: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
```
checking bpftrace code:
```code
[tkloczko@pers-jacek bpftrace-0.18.0]$ grep -r gzgets
src/utils.cpp:    // gzopen/gzgets handle both uncompressed and compressed files
src/utils.cpp:    while (gzgets(file, buf, sizeof(buf)))
```
So it uses zlib API.
Looks like bpftrace build is tring to use indirect linking.

tested after build failed
```console
[ 33%] Linking CXX executable bpftrace
cd /home/tkloczko/rpmbuild/BUILD/bpftrace-0.18.0/x86_64-redhat-linux-gnu/src && /usr/bin/cmake -E cmake_link_script CMakeFiles/bpftrace.dir/link.txt --verbose=1
/usr/bin/g++ -O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -flto=auto -flto-partition=none -Os -DNDEBUG -Wl,--gc-sections -Wl,--as-needed -flto=auto -flto-partition=none -fuse-linker-plugin -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -Wl,--build-id=sha1 CMakeFiles/bpftrace.dir/main.cpp.o -o bpftrace  libbpftrace.a ../resources/libresources.a libruntime.a /usr/lib64/libbpf.so /usr/lib64/libbcc_bpf.so /usr/lib64/libelf.so aot/libaot.a ast/libast.a ../libparser.a ast/libast_defs.a /usr/lib64/libclang.so.16.0.3 /usr/lib64/libLLVM-16.so.0 arch/libarch.a cxxdemangler/libcxxdemangler_llvm.a
/home/tkloczko/rpmbuild/BUILD/bpftrace-0.18.0/src/driver.cpp:8:14: warning: ‘yy_scan_string’ violates the C++ One Definition Rule [-Wodr]
    8 | extern void *yy_scan_string(const char *yy_str, yyscan_t yyscanner);
      |              ^
/home/tkloczko/rpmbuild/BUILD/bpftrace-0.18.0/x86_64-redhat-linux-gnu/lex.yy.cc:2560:17: note: ‘yy_scan_string’ was previously declared here
 2560 |  */
      |                 ^
/usr/bin/ld: /usr/lib64/libbcc_bpf.so: undefined reference to symbol 'gzgets'
/usr/bin/ld: /usr/lib64/libz.so.1: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [src/CMakeFiles/bpftrace.dir/build.make:114: src/bpftrace] Error 1
make[2]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/bpftrace-0.18.0/x86_64-redhat-linux-gnu'
make[1]: *** [CMakeFiles/Makefile2:1388: src/CMakeFiles/bpftrace.dir/all] Error 2
make[1]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/bpftrace-0.18.0/x86_64-redhat-linux-gnu'
make: *** [Makefile:149: all] Error 2
```
I've manually executed
```console
[tkloczko@pers-jacek x86_64-redhat-linux-gnu]$ cd src; /usr/bin/g++ -O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -flto=auto -flto-partition=none -Os -DNDEBUG -Wl,--gc-sections -Wl,--as-needed -flto=auto -flto-partition=none -fuse-linker-plugin -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -Wl,--build-id=sha1 CMakeFiles/bpftrace.dir/main.cpp.o -o bpftrace  libbpftrace.a ../resources/libresources.a libruntime.a /usr/lib64/libbpf.so /usr/lib64/libbcc_bpf.so /usr/lib64/libelf.so aot/libaot.a ast/libast.a ../libparser.a ast/libast_defs.a /usr/lib64/libclang.so.16.0.3 /usr/lib64/libLLVM-16.so.0 arch/libarch.a cxxdemangler/libcxxdemangler_llvm.a -lz
/home/tkloczko/rpmbuild/BUILD/bpftrace-0.18.0/src/driver.cpp:8:14: warning: ‘yy_scan_string’ violates the C++ One Definition Rule [-Wodr]
    8 | extern void *yy_scan_string(const char *yy_str, yyscan_t yyscanner);
      |              ^
/home/tkloczko/rpmbuild/BUILD/bpftrace-0.18.0/x86_64-redhat-linux-gnu/lex.yy.cc:2560:17: note: ‘yy_scan_string’ was previously declared here
 2560 |  */
      |                 ^
[tkloczko@pers-jacek src]$
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -110,6 +110,9 @@ set_property( GLOBAL PROPERTY FIND_LIBRARY_USE_LIB64_PATHS TRUE )
 
 include_directories(SYSTEM ${KERNEL_INCLUDE_DIRS})
 
+find_package(ZLIB REQUIRED)
+include_directories(SYSTEM ${ZLIB_INCLUDE_DIRS})
+
 find_package(LibBcc REQUIRED)
 include_directories(SYSTEM ${LIBBCC_INCLUDE_DIRS})
 

--- a/cmake/FindLibBfd.cmake
+++ b/cmake/FindLibBfd.cmake
@@ -52,7 +52,7 @@ SET(CMAKE_REQUIRED_LIBRARIES ${LIBBFD_LIBRARIES} ${LIBOPCODES_LIBRARIES})
 # do it. Furthermore, libbfd uses some libc symbols that we must manually
 # link against if we're not using static libc (which includes such symbols).
 if(STATIC_LINKING)
-  find_package(LibZ)
+  find_package(ZLIB)
   list(APPEND CMAKE_REQUIRED_LIBRARIES ${LIBIBERTY_LIBRARIES} ${LIBZ_LIBRARIES})
   if(NOT STATIC_LIBC)
     set(CMAKE_REQUIRED_FLAGS

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -125,7 +125,7 @@ if(STATIC_LINKING)
 endif(STATIC_LINKING)
 
 
-target_link_libraries(runtime ${LIBBPF_LIBRARIES})
+target_link_libraries(runtime ${LIBBPF_LIBRARIES} ${ZLIB_LIBRARIES})
 target_link_libraries(libbpftrace parser resources runtime aot ast arch cxxdemangler_llvm)
 
 if(LIBPCAP_FOUND)

